dipy/align/tests/test_imaffine.py

```diff
@@ -0,0 +1,251 @@
+                        static_grid2space, moving, moving_grid2space)
+                    assert_array_almost_equal(actual, expected)
+
+
+def test_aff_origins_3d():
+    # Create arbitrary image-to-space transforms
+    axis = np.array([.5, 2.0, 1.5])
+    t = 0.15 #translation factor
+
+    for theta in [-1 * np.pi/6.0, 0.0, np.pi/5.0]: #rotation angle
+        for s in [0.83,  1.3, 2.07]: #scale
+            m_shapes = [(256, 256, 128), (255, 255, 127), (64, 127, 142)]
+            for shape_moving in m_shapes:
+                s_shapes = [(256, 256, 128), (255, 255, 127), (64, 127, 142)]
+                for shape_static in s_shapes:
+                    moving = np.ndarray(shape=shape_moving)
+                    static = np.ndarray(shape=shape_static)
+                    trans = np.array([[1, 0, 0, -t*shape_static[0]],
+                                      [0, 1, 0, -t*shape_static[1]],
+                                      [0, 0, 1, -t*shape_static[2]],
+                                      [0, 0, 0, 1]])
+                    trans_inv = np.linalg.inv(trans)
+                    rot = np.zeros(shape=(4,4))
+                    rot[:3, :3] = geometry.rodrigues_axis_rotation(axis, theta)
+                    rot[3,3] = 1.0
+                    scale = np.array([[1*s, 0, 0, 0],
+                                      [0, 1*s, 0, 0],
+                                      [0, 0, 1*s, 0],
+                                      [0, 0, 0, 1]])
+
+                    static_grid2space = trans_inv.dot(scale.dot(rot.dot(trans)))
+                    moving_grid2space = np.linalg.inv(static_grid2space)
+
+                    # Expected translation
+                    c_static = static_grid2space[:3, 3]
+                    c_moving = moving_grid2space[:3, 3]
+                    expected = np.eye(4);
+                    expected[:3, 3] = c_moving - c_static
+
+                    # Implementation under test
+                    actual = imaffine.aff_origins(static, static_grid2space,
+                                                  moving, moving_grid2space)
+                    assert_array_almost_equal(actual, expected)
+
+
+def test_affreg_all_transforms():
+    # Test affine registration using all transforms with typical settings
+    for ttype in factors.keys():
+        dim = ttype[1]
+        if dim == 2:
+            nslices = 1
+        else:
+            nslices = 45
+        factor = factors[ttype][0]
+        sampling_pc = factors[ttype][1]
+        transform = regtransforms[ttype]
+
+        static, moving, static_grid2space, moving_grid2space, smask, mmask, T = \
+                        setup_random_transform(transform, factor, nslices, 1.0)
+        # Sum of absolute differences
+        start_sad = np.abs(static - moving).sum()
+        metric = imaffine.MattesMIMetric(32, sampling_pc)
+        affreg = imaffine.AffineRegistration(metric,
+                                             [10000, 1000, 100], 1e-5,
```

Comment:
These are just settings to obtain a "good" registration so we can verify that the final alignment is good. Let me think what other settings would be interesting to test ("malicious" settings like lists of strings and things like that, maybe?).